lib/puppet/pops/types/types.rb

```diff
@@ -714,7 +714,7 @@ def roundtrip_with_string?
 
   # @api private
   def _assignable?(o, guard)
-    o.is_a?(PScalarType) ||
+    o.instance?(PScalarType) ||
```

Comment:
I think this needs to be `is_a?` since `PNumericType` (and others) extend `PScalarDataType` and that extends `PScalarType`.